ipu.py

```diff
@@ -442,15 +442,38 @@ def start(self) -> None:
     def cold_boot(self) -> None:
         pass
 
-    def version(self) -> str:
+    def _version_via_redfish(self) -> Optional[str]:
+        url = f"https://{self.url}:8443/redfish/v1/Managers/1"
+        res = self._run_curl(f"-k '{url}'")
+        if res.returncode != 0:
+            return None
+        try:
+            data = json.loads(res.out)
+        except Exception:
+            return None
+        fwversion = data.get("FirmwareVersion")
+        if not isinstance(fwversion, str):
+            return None
+        match = re.search(r"^MEV-.*\.([0-9]+\.[0-9]+\.[0-9]+)\.[0-9]+$", fwversion.strip())
+        if not match:
+            return None
+        return match.group(1)
+
+    def _version_via_ssh(self) -> Optional[str]:
         rh = host.RemoteHost(self.url)
         rh.ssh_connect("root", password="", discover_auth=False)
         contents = rh.read_file("/etc/issue")
         match = re.search(r"Version: (\S+)", contents)
-        if match is None:
-            sys.exit(-1)
+        if not match:
+            return None
         return match.group(1).strip()
 
+    def version(self) -> str:
+        fwversion = self._version_via_redfish() or self._version_via_ssh()
```

Comment:
Final comment after thinking about it more: If we can't get the version through redfish, then we need to error out. We want it to fail if it's impossible to get the version from redfish.

Reply:
dropped SSH way.